src/v/cluster/scheduling/leader_balancer_greedy.h

```diff
@@ -193,6 +208,7 @@ class greedy_balanced_shards final : private leader_balancer_strategy {
     size_t _num_cores;
     size_t _num_groups;
     std::vector<index_type::const_iterator> _load;
+    absl::flat_hash_map<model::broker_shard, size_t> _load_map;
```

Comment:
Flat_hash_map should be OK in terms of not allocating too much contiguous memory here as this map is only size of the number of shards, so this should be in the 1000s currently. I calculate 1000 shards would need a 24K byte backing flat_map.